Watchman.Discord/Areas/Protection/Controllers/MuteUserController.cs

```diff
@@ -14,35 +20,40 @@ public class MuteUserController : IController
         private readonly MutingService _mutingService;
         private readonly UnmutingService _unmutingService;
         private readonly UsersService _usersService;
+        private readonly MessagesServiceFactory _messagesServiceFactory;
 
-        public MuteUserController(MutingService mutingService, UnmutingService unmutingService, UsersService usersService)
+        public MuteUserController(MutingService mutingService, UnmutingService unmutingService, UsersService usersService, MessagesServiceFactory messagesServiceFactory)
         {
             this._mutingService = mutingService;
             this._unmutingService = unmutingService;
             this._usersService = usersService;
+            this._messagesServiceFactory = messagesServiceFactory;
         }
 
-        [DiscordCommand("mute")]
         [AdminCommand]
-        public async Task MuteUser(DiscordRequest request, Contexts contexts)
+        public async Task MuteUser(MuteCommand command, Contexts contexts)
         {
-            var requestParser = new MuteRequestParser(request, this._usersService, contexts);
-            var userToMute = requestParser.GetUser();
-            if (userToMute.Id == this._usersService.GetBot().Id)
+            var userToMute = await this._usersService.GetUserByIdAsync(contexts.Server, command.User);
+            if (userToMute == null)
             {
-                throw new UserDidntMentionAnyUserException();
```

Comment:
myślę, że lepiej rzucać wyjątek, z tą różnicą, że UserNotFound i tam przekazuj user mention

\+ @Rafal38 zrobił extension method pod ulong.GetUserMention
ale to zależy kto szybciej skończy całość, więc na razie zostaw ten komentarz otwarty i poźniej przed samym mergem zobaczymy czy będziesz już mógł z tamtego brancha użyć tej metody